symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/symphony/HealthCheckClientImplTest.java

```diff
@@ -0,0 +1,100 @@
+package com.symphony.bdk.bot.sdk.symphony;
+
+import static org.junit.Assert.assertEquals;
+
+import com.symphony.bdk.bot.sdk.lib.restclient.RestClient;
+import com.symphony.bdk.bot.sdk.lib.restclient.model.RestResponse;
+import com.symphony.bdk.bot.sdk.symphony.model.HealthCheckInfo;
+
+import clients.SymBotClient;
+import clients.symphony.api.HealthcheckClient;
+import configuration.SymConfig;
+import configuration.SymConfigLoader;
+import model.HealthcheckResponse;
+import org.apache.commons.io.serialization.ClassNameMatcher;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Value;
+
+public class HealthCheckClientImplTest {
+
+  private SymBotClient symBotClient;
+  private RestClient restClient;
+  private HealthcheckResponse healthcheckResponse;
+  private HealthcheckClientImpl healthcheckClientImpl;
+
+  @Value("${management.symphony-api-client-version}")
+  private String symphonyApiClientVersion;
+
+  @Before
+  public void initBot() {
+    this.initHealthCheckResponse();
+
+    final SymConfig symConfig = SymConfigLoader.loadConfig("src/test/resources/sym-config.json");
+
+    final HealthcheckClient healthcheckClient = Mockito.mock(HealthcheckClient.class);
+    Mockito.when(healthcheckClient.performHealthCheck()).thenReturn(healthcheckResponse);
+
+    this.symBotClient = Mockito.mock(SymBotClient.class);
+    Mockito.when(this.symBotClient.getConfig()).thenReturn(symConfig);
+    Mockito.when(this.symBotClient.getHealthcheckClient()).thenReturn(healthcheckClient);
+
+    this.initRestClient();
+
+    this.healthcheckClientImpl = new HealthcheckClientImpl(this.symBotClient, this.restClient);
+  }
+
+  @Test
+  public void testHealthCheck() {
+    final HealthCheckInfo healthCheckInfoExpected = this.initHealthCheckInfo();
+
+    final HealthCheckInfo healthCheckInfo = this.healthcheckClientImpl.healthCheck();
+    assertEquals(healthCheckInfoExpected, healthCheckInfo);
+  }
+
+  @Test(expected = Exception.class)
+  public void testHealthCheckWithException(){
+    Mockito.when(this.symBotClient.getHealthcheckClient().performHealthCheck()).thenThrow(Exception.class);
+    this.healthcheckClientImpl.healthCheck();
+  }
+
+  @Test(expected = Exception.class)
+  public void testInitRestClientWithException(){
+    Mockito.when(this.restClient.getRequest(ArgumentMatchers.anyString(), ClassNameMatcher.class)).thenThrow(Exception.class);
+    this.restClient.getRequest("https://localhost/pod/v1/podcert", String.class);
+  }
+
+  private void initRestClient() {
+    this.restClient = Mockito.mock(RestClient.class);
+
+    final RestResponse<String> restResponse = new RestResponse<>();
+
+    Mockito.when(this.restClient.getRequest(
+        "https://localhost/pod/v1/podcert", String.class
+    )).thenReturn(restResponse);
```

Comment:
To throw an exception inside private method `checkPodStatus`, you have to modify that with `thenThrow(new Exception())`

Reply:
I've just repushed but the coverage doesn't change

Reply:
The coverage has increased with the failure tests
![coverage_184](https://user-images.githubusercontent.com/68892288/103638325-0e8b0500-4f4d-11eb-8637-de37dd6eca98.JPG)